rest_framework/views.py

```diff
@@ -555,7 +556,16 @@ def options(self, request, *args, **kwargs):
         """
         Handler method for HTTP 'OPTIONS' request.
         """
-        if self.metadata_class is None:
-            return self.http_method_not_allowed(request, *args, **kwargs)
-        data = self.metadata_class().determine_metadata(request, self)
-        return Response(data, status=status.HTTP_200_OK)
+        def func():
+            if self.metadata_class is None:
+                return self.http_method_not_allowed(request, *args, **kwargs)
+            data = self.metadata_class().determine_metadata(request, self)
+            return Response(data, status=status.HTTP_200_OK)
+
+        if hasattr(self, 'view_is_async') and self.view_is_async:
```

Comment:
`getattr(self, 'view_is_async', False)` ?

Reply:
Agreed, could be simplified into one conditional if using `getattr`